### PR TITLE
[FIX] Fixed scrolling for long lists

### PR DESCRIPTION
**Pull request by stopdrake**

### CODE REVIEW

**Review by kanavarora — APPROVED** [review]

File: checkout-cart/src/main/java/com/urbanclap/checkoutcart/market_impl/recycler_view_market/RecyclerMarketManager.java [hunk 1]
```diff
@@ -103,16 +103,25 @@ public void onScrollPosition(int pos) {
...
                 break;
             }
         }
-
-        if (pos != -1) {
-            smoothScroller.setTargetPosition(pos);
+        if (smoothScrollPos != -1) {
+            int visiblePosition = ((LinearLayoutManager)layoutManager).findFirstVisibleItemPosition();
+            int scrollPos = -1;
+            if(smoothScrollPos - 10 > visiblePosition)
```
Comment (kanavarora) [inline checkout-cart/src/main/java/com/urbanclap/checkoutcart/market_impl/recycler_view_market/RecyclerMarketManager.java]: ideally this 10 should be a constant somewhere

File: checkout-cart/src/main/java/com/urbanclap/checkoutcart/market_impl/recycler_view_market/RecyclerMarketManager.java [hunk 2]
```diff
@@ -103,16 +103,25 @@ public void onScrollPosition(int pos) {
...
+                scrollPos = smoothScrollPos - 10;
+            else if(smoothScrollPos + 10 < visiblePosition)
```
Comment (kanavarora) [inline checkout-cart/src/main/java/com/urbanclap/checkoutcart/market_impl/recycler_view_market/RecyclerMarketManager.java]: space missing after if

Comment (kanavarora) [inline checkout-cart/src/main/java/com/urbanclap/checkoutcart/market_impl/recycler_view_market/RecyclerMarketManager.java]: space missing after if

File: checkout-cart/src/main/java/com/urbanclap/checkoutcart/market_impl/recycler_view_market/RecyclerMarketManager.java [hunk 3]
```diff
@@ -103,16 +103,25 @@ public void onScrollPosition(int pos) {
...
+                scrollPos = smoothScrollPos + 10;
+
+            if(scrollPos != -1)
```
Comment (kanavarora) [inline checkout-cart/src/main/java/com/urbanclap/checkoutcart/market_impl/recycler_view_market/RecyclerMarketManager.java]: space missing after if

**Comment by stopdrake** [comment]
@kanavarora updated review change

**Pull request merged**